lib/api/src/modules/stories.ts

```diff
@@ -268,10 +268,17 @@ export const init: ModuleFn = ({
     // Later when we change story via the manager (or SELECT_STORY below), we'll already be at the
     // correct path before CURRENT_STORY_WAS_SET is emitted, so this is less important (the navigate is a no-op)
     // Note this is the case for refs also.
-    fullAPI.on(CURRENT_STORY_WAS_SET, function handleCurrentStoryWasSet({ storyId, viewMode }) {
+    fullAPI.on(CURRENT_STORY_WAS_SET, async function handleCurrentStoryWasSet({
+      storyId,
+      viewMode,
+    }) {
       const { source }: { source: string } = this;
       const [sourceType] = getSourceType(source);
 
+      if ((await fullAPI.showReleaseNotesOnLaunch()) && !fullAPI.didViewReleaseNotes()) {
+        return;
+      }
```

Comment:
Should this be more general? I can't imagine it is a good idea to close any settings page randomly when the SB loads.

Reply:
Yeah, it would be nice to generalize. I think the hard part about that is that this callback handles all events for `CURRENT_STORY_WAS_SET` and therefore I can't just say "if on the settings page, don't redirect" because you might be intentionally navigating away from the settings page. Thoughts?

Reply:
Should we make it the direct action that closes the settings, rather than the side effect? 

E.g close the settings when you click the close button or a story in the sidebar, instead of when the story change happens in the sidebar. 

I actually think this’ll happen pretty naturally if you block the routing here when settings is open. Here’s my thinking. There are two ways this event can fire:

1. Preview boots (at the very start OR on hot module reload)

2. In response to the manager selecting a new story

This code is here only for case 1, as:

If we break down what happens in 2:

A. Something triggers `SELECT_STORY`
B. That makes the manager change route (this would close settings)
C. Manager emits `SET_CURRENT_STORY` in response to route change
D. Preview changes story, emits `CURRENT_STORY_WAS_SET` which hits this code path
E. We try and route again to where we already are, which is a no-op

It’s a bit confusing for sure but hopefully that helps?

Reply:
Updated w/ a reference to the new settings module to determine if it is open. Seems to work well (though I have all of those `setTimeout` navigate calls commented out) -- I just need to circle back once those are removed.